docs/DeveloperGuide.md

```diff
@@ -1441,24 +1441,54 @@ Deleting sessions in the students tab
 
 { end of `manual_testing#deleting_a_session` written by: Marcus Tan Wei }
 
-{ start of `appendix` written by: _________ }
+{ start of `appendix` written by: Masagca Eris Jacey }
 
 ## 12. Appendix: effort
 
 **Difficulty level:**
 
-* todo
+The overall difficulty level of the project was relatively moderate. 
+For most of our team, this was our first significant foray into the world of software engineering, and specifically
+our first time working in a group to develop a medium-sized application.
 
 **Challenges faced:**
 
-* todo
+* It was our first time working on a brown-field project, so a few weeks were needed just to get used to the existing codebase.
+* It was also our first time dealing with some of the libraries and frameworks used in this project (like Jackson), 
+  so a bit of time was needed to be familiar with them.
+* (in the beginning especially) We faced issues regarding project management (specifically role delegation), and
+  we had troubles with resolving merge conflicts and meeting deadlines in our first iteration.
+* With each new iteration, we would sometimes face regression issues, as some new features would cause older ones to be buggy. 
+* AB3 only dealt with a single entity type, but for **ATAS**, we had to deal with multiple entity types (students, 
+  sessions, attributes) which had their own specific set of complications and subtle differences in implementations.  
+* Significant time had to be spent on fixing and refining the following features: 
+   * "current session": This was an integration feature done during an early iteration when we were still relatively unfamiliar with the codebase.
+   * "undo/redo": This was a feature that was hard to get right because of the different entity types being slightly different in nature.
+     ("students" were a "one-layer" entity while each "session" had their own list of "attributes", making it a "two-layer" entity)
 
 **Effort required:**
 
-* todo
+* Weekly meetings were held (about 2 hours every Wednesday) to review what had been done in the week prior, as well as 
+  discuss future tasks for the next week.
+* Our team worked tirelessly week in and week out for several (~7) weeks, sparing no effort during each iteration 
+  (on both new implementations as well as making sure the older ones aren't buggy).
```

Comment:
Minor nitpick but should aren't be are not? And emphasized -> emphasised, noticed in other areas as well

Reply:
I think emphasize is the more common version, at least in this module we seem to be following American conventions (color, etc.)